flink-core/src/main/java/org/apache/flink/metrics/MetricRegistry.java

```diff
@@ -64,6 +66,15 @@ public MetricRegistry(Configuration config) {
 		}
 		this.scopeFormats = scopeFormats;
 
+		char delim;
+		try {
+			delim = config.getString(ConfigConstants.METRICS_SCOPE_DELIMITER, ".").charAt(0);
```

Comment:
Sorry, it was a hick-up of Github, because it didn't show me that I posted the comment so I thought I clicked on cancel.

Why not changing `ScopeFormat.concat` such that it always takes a delimiter string as first parameter? Giving `""` as first argument will lead to the former variant where you don't specify a delimiter.